### PR TITLE
Addtional fixes for easybuild-easyblocks#3340

### PR DESCRIPTION
**Pull request by lexming**

### CODE REVIEW

File: easybuild/easyblocks/o/openssl_wrapper.py [hunk 1]
```diff
@@ -288,20 +282,23 @@ def __init__(self, *args, **kwargs):
...
-            self.system_ssl['bin'] = which('openssl')
+        target_ssl_bins = ['openssl']
+        if self.generation == '1.1':
+            target_ssl_bins.insert(0, 'openssl11')  # prefer 'openssl11' over 'openssl' with v1.1
+        elif self.generation == '3':
+            target_ssl_bins.insert(0, 'openssl3')  # prefer 'openssl3' over 'openssl' with v3
+
+        for target_bin in target_ssl_bins:
+            target_bin_path, target_bin_version = self.get_openssl_bin_version(target_bin)
+            if target_bin_version == self.system_ssl['version']:
```
Comment (SebastianAchilles) [inline easybuild/easyblocks/o/openssl_wrapper.py]: This check is too strict for Ubuntu.

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/o/openssl_wrapper.py (hunk 1 above)
Comment (lexming) [reply easybuild/easyblocks/o/openssl_wrapper.py]: Ok, changed approach. Now we define a minimum **and** maximum version and only check that each component falls in that version range. Components may have different versions between them.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _OpenSSL-3.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
zen2-ubuntu-2404 - Linux Ubuntu 24.04 LTS (Noble Numbat), x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.12.3
See https://gist.github.com/SebastianAchilles/401896cf14f25d9a11e14e2bf125bf11 for a full test report.

File: easybuild/easyblocks/o/openssl_wrapper.py (hunk 1 above)
Comment (SebastianAchilles) [reply easybuild/easyblocks/o/openssl_wrapper.py]: Thanks a lot :+1: It is working now on Ubuntu 24.04 and 22.04.

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Pull request merged**